core/state/statedb.go

```diff
@@ -158,14 +161,14 @@ func New(root common.Hash, db Database, snaps *snapshot.Tree) (*StateDB, error)
 	return sdb, nil
 }
 
-// TODO ricardolyn: separate commit
+// Quorum
 // NewDual - Create a public and private state from a given public and private tree
-func NewDual(root common.Hash, db Database, snaps *snapshot.Tree, privateRoot common.Hash, privateDb Database, privateSnaps *snapshot.Tree) (*StateDB, *StateDB, error) {
+func NewDual(root common.Hash, db Database, snaps *snapshot.Tree, ethDb ethdb.Database, privateDb Database, privateSnaps *snapshot.Tree) (*StateDB, *StateDB, error) {
```

Comment:
MINOR: Function signature would be clearer/cleaner as follow:

```go
func NewDual(root common.Hash, ethDb ethdb.Database, db, privateDb Database, snaps, privateSnaps *snapshot.Tree) (state, privateState *StateDB, err error)
```

Reply:
agree with most, but there is one point about `root common.Hash, db Database, snaps *snapshot.Tree`. I didn't want to change the order of those arguments to match the same start as `New(...)`. what do you think?

Reply:
Ok